core/spec/models/spree/unit_cancel_spec.rb

```diff
@@ -38,21 +38,8 @@
 
     context "all inventory on the line item are not canceled" do
       it "divides the line item total by the inventory units size" do
-        expect(subject).to eq(-5.0)
-      end
-    end
-
-    context "some inventory on the line item is canceled" do
-      before { inventory_unit2.cancel! }
-
-      it "divides the line item total by the uncanceled units size" do
         expect(subject).to eq(-10.0)
       end
-
-      it "raises an error if dividing by 0" do
```

Comment:
Should we preserve the behaviour where this fails if all units are already cancelled? Either way, I think we should have a spec for that behaviour.

Reply:
I think recalculating the amount should not fail, but adjusting the line item should. OTOH, `OrderCancellations` will already fail if the inventory unit is already canceled: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order_cancellations.rb#L108 (in a transaction, will fail with an invalid state transition. Not sure we should add a spec to the recalculator.